src/main/java/com/github/manolo8/darkbot/gui/nearplayers/NearPlayerRenderer.java

```diff
@@ -0,0 +1,87 @@
+package com.github.manolo8.darkbot.gui.nearplayers;
+
+import com.github.manolo8.darkbot.config.PlayerInfo;
+import com.github.manolo8.darkbot.config.PlayerTag;
+import com.github.manolo8.darkbot.gui.utils.UIUtils;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import javax.swing.border.Border;
+import javax.swing.border.CompoundBorder;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class NearPlayerRenderer extends JPanel implements ListCellRenderer<PlayerInfo> {
```

Comment:
I'm pretty sure you could just extend the default renderer for this one, you're not displaying tags or anything too complex in this list, rewriting an impl is a bit overkill, you could simply display the username on this, no need for the id since it's all ephermal